src/ServerRequestFactory.php

```diff
@@ -198,21 +198,18 @@ public static function marshalHeaders(array $server)
     {
         $headers = [];
         foreach ($server as $key => $value) {
-            if ($value && strpos($key, 'HTTP_') === 0) {
-                $name = strtr(substr($key, 5), '_', ' ');
-                $name = strtr(ucwords(strtolower($name)), ' ', '-');
+            if ($value) {
```

Comment:
This code has a much higher cyclomatic complexity, and is also harder to understand, IMO